s3/pkg/db/drivers/mongo/listbuckets.go

```diff
@@ -15,22 +15,30 @@
 package mongo
 
 import (
+	"context"
+
 	"github.com/globalsign/mgo/bson"
 	"github.com/micro/go-log"
 	. "github.com/opensds/multi-cloud/s3/pkg/exception"
 	pb "github.com/opensds/multi-cloud/s3/proto"
 )
 
-func (ad *adapter) ListBuckets(in *pb.BaseRequest, out *[]pb.Bucket) S3Error {
+func (ad *adapter) ListBuckets(ctx context.Context, in *pb.BaseRequest, out *[]pb.Bucket) S3Error {
 	ss := ad.s.Copy()
 	defer ss.Close()
 	c := ss.DB(DataBaseName).C(BucketMD)
 
-	log.Log("Find buckets from database...... \n")
+	log.Log("list buckets from database...... \n")
+
+	m := bson.M{}
+	err := UpdateContextFilter(ctx, m)
+	if err != nil {
+		return InternalError
+	}
 
-	err := c.Find(bson.M{"owner": in.Id}).All(out)
+	err = c.Find(m).All(out)
 	if err != nil {
-		log.Log("Find buckets from database failed, err:%v\n", err)
+		log.Log("find buckets from database failed, err:%v\n", err)
 		return InternalError
```

Comment:
In furture can we design to have better errors. more granular. like in this case DBInternalError

Reply:
fixed